crates/rust-project-goals-cli/src/cfp.rs

```diff
@@ -0,0 +1,533 @@
+use std::fs::{self, File};
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use anyhow::{Context, Result};
+use regex::Regex;
+
+/// Module containing pure functions for text processing
+pub mod text_processing {
+    use regex::Regex;
+    
+    /// Process template content by replacing placeholders and removing notes
+    pub fn process_template_content(content: &str, timeframe: &str, lowercase_timeframe: &str) -> String {
+        // Remove note sections (starting with '> **NOTE:**' and continuing until a non-'>' line)
+        // But preserve other content starting with '>'
+        let lines: Vec<&str> = content.lines().collect();
+        let mut filtered_lines = Vec::new();
+        let mut in_note_section = false;
+        
+        for line in lines.iter() {
+            if line.trim().starts_with("> **NOTE:**") {
+                in_note_section = true;
+                continue;
+            }
+            
+            if in_note_section {
+                if line.trim().starts_with(">") {
+                    continue; // Skip this line as it's part of the note section
+                } else {
+                    in_note_section = false; // End of note section
+                }
+            }
+            
+            filtered_lines.push(*line);
+        }
+        
+        // Join the filtered lines and clean up consecutive empty lines
+        let mut cleaned_lines = Vec::new();
+        let mut last_was_empty = false;
+        
+        for line in filtered_lines {
+            let is_empty = line.trim().is_empty();
+            if !is_empty || !last_was_empty {
+                cleaned_lines.push(line);
+            }
+            last_was_empty = is_empty;
+        }
+        
+        // Join and process placeholders
+        cleaned_lines.join("\n")
+            .replace("YYYYHN", timeframe)
```

Comment:
(This is a note for the things we can fix that is discussed elsewhere: to me this is correct, but this function assumes `timeframe` is uppercase while the others do not)